tcms/kiwi_auth/admin.py

```diff
@@ -144,6 +144,29 @@ def delete_model(self, request, obj):
         delete_user(obj)
 
 
+class KiwiGroupAdmin(GroupAdmin):
+
+    def has_delete_permission(self, request, obj=None):
+        if obj and obj.name in ['Tester', 'Administrator']:
+            return False
+        return super().has_delete_permission(request, obj)
+
+    def get_fields(self, request, obj=None):
+        form = self._get_form_for_get_fields(request, obj)
+        # set the correct order of the fields in the UI
+        return [*self.get_readonly_fields(request, obj), *form.base_fields]
```

Comment:
You aren't replicating the parent method exactly and in general I don't like changing methods internals b/c Django can change them underneath us and lead to hard to debug problems. Instead use this:

```
    def get_fields(self, request, obj=None):
        fields = super().get_fields(request, obj=obj)
        name_index = fields.index('name')

        # make sure Name is always the first field
        if name_index > 0:
            del fields[name_index]
            fields.insert(0, 'name')

        return fields
```

which relies on the parent behavior and just makes sure fields are always in the order we want.